main.tf

```diff
@@ -59,6 +66,15 @@ resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_crn" {
   }
 }
 
+data "ibm_is_vpc" "vpc" {
+  count = var.create_vpc == false ? 1 : 0
+  id    = var.existing_vpc_id
```

Comment:
I think you will need to specify the resource group here, meaning you need a new inout variable called `existing_resource_group_id`

Reply:
There is no mention of `resource_group` being a parameter for data source
https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/data-sources/is_vpc